store/metadata.go

```diff
@@ -213,6 +232,10 @@ func (ms *MetadataStore) getLinkedPeersInfo() (map[string]bool, []metadata.Conta
 		linkedFromServices := ms.getLinkedFromServicesToSelf()
 		for _, aService := range linkedFromServices {
 			for _, aContainer := range aService.Containers {
+				// Skip containers whose network names don't match self
+				if ms.info.networksMap[aContainer.NetworkUUID].Name != ms.info.selfNetwork.Name {
```

Comment:
yes this is needed because, when the vxlan stacks are linked, their uuids are different but with the same name.